inc/shortcodes/class-shortcode.php

```diff
@@ -55,4 +55,61 @@ public static function callback( $attrs, $content = '' ) {
 		return '';
 	}
 
+	/**
+	 * Parse iframes from a string, if there are any
+	 *
+	 * @param string $content
+	 * @return array|false
+	 */
+	protected static function parse_iframes( $content ) {
+
+		if ( false === stripos( $content, '<iframe' ) ) {
+			return false;
+		}
+
+		if ( preg_match_all( '#(.+\r?\n?)?(<iframe([^>]+)>([^<]+)?</iframe>)(\r?\n?.+)?#', $content, $matches ) ) {
+			$iframes = array();
+			foreach ( $matches[0] as $key => $value ) {
+				$iframe = new \stdClass;
+				$iframe->original = $matches[2][ $key ];
+				$iframe->before = $matches[1][ $key ];
+				$iframe->attrs = array( 'src' => '' );
+				$iframe->inner = $matches[4][ $key ];
+				$iframe->after = $matches[5][ $key ];
+				$parts = explode( ' ', $matches[3][ $key ] );
```

Comment:
What does this do if a space appears inside an attribute value, like a list of class names or a title?